workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs

```diff
@@ -12,38 +12,39 @@ namespace Improbable.Gdk.GameObjectCreation
     ///     and links it to the entity via the EntityGameObjectLinker. Also checks for entity removal and calls the
     ///     IEntityGameObjectCreator for cleanup.
     /// </summary>
-    [UpdateInGroup(typeof(SpatialOSReceiveGroup.EntityInitialisationGroup))]
     [DisableAutoCreation]
     internal class GameObjectInitializationSystem : ComponentSystem
     {
-        // TODO: Add our ISystemStateComponentData for all entities instead of ones with GameObjects, store spatial ID
-        // callbacks instead of in it, give that in removal callback instead of SpatialOSEntity struct.
+        public struct HasGameObjectSystemState : ISystemStateComponentData
+        {
+            public SpatialEntityId EntityId;
+        }
 
         private struct AddedEntitiesData
         {
             public readonly int Length;
-            public EntityArray Entities;
+            [ReadOnly] public EntityArray Entities;
             [ReadOnly] public ComponentDataArray<SpatialEntityId> SpatialEntityIds;
-            [ReadOnly] public ComponentDataArray<NewlyAddedSpatialOSEntity> NewlyCreatedEntities;
+            [ReadOnly] public SubtractiveComponent<HasGameObjectSystemState> NoHasGameObjectSystemStates;
         }
 
         private struct RemovedEntitiesData
         {
             public readonly int Length;
-            public EntityArray Entities;
-            [ReadOnly] public ComponentDataArray<GameObjectReferenceHandle> GameObjectReferenceHandles;
-            public SubtractiveComponent<GameObjectReference> NoGameObjectReference;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<HasGameObjectSystemState> HasGameObjectSystemStates;
+            [ReadOnly] public SubtractiveComponent<SpatialEntityId> NoSpatialEntityIds;
         }
 
         [Inject] private AddedEntitiesData addedEntitiesData;
         [Inject] private RemovedEntitiesData removedEntitiesData;
 
-        [Inject] private EntityGameObjectLinkerSystem LinkerSystem;
+        [Inject] private EntityGameObjectLinkerSystem linkerSystem;
         [Inject] private WorkerSystem worker;
 
         private ViewCommandBuffer viewCommandBuffer;
-        private readonly Dictionary<Entity, GameObject> entityGameObjectCache = new Dictionary<Entity, GameObject>();
         private readonly IEntityGameObjectCreator gameObjectCreator;
+        private readonly Dictionary<Entity, GameObject> entityToGameObjects = new Dictionary<Entity, GameObject>();
```

Comment:
nitpick - private readonly should go above non readonly. Injected is probably a special case that goes wherever it seems best.